### PR TITLE
Fix bug when checking if a JobSet is active during tests.

### LINKED ISSUE #530: Bug in testing utility function JobSetActive
@jedwins1998 encountered a bug when working on tests for configurable failure policy changes, tests would pass when using `JobSetFailed()` to check the JobSet has a failed condition type with a status of true, but also would pass when `JobSetFailed()` was changed to `JobSetActive()`. 

I think this is because in [JobSetActive](https://github.com/kubernetes-sigs/jobset/blob/9dc4358670f119164517f90d6005bd4742a4ff5d/test/util/util.go#L120) we are simply passing in an empty list of conditions to checkJobSetStatus, which verifies all given conditions are set on the jobset status. 

However, even for a failed JobSet, the number of conditions matching an empty list of conditions will be 0, and the check will return ok, which it should not.

Instead, we should change JobSetActive to verify there are no mutually exclusive conditions set on the JobSet status, like failed or completed.

### PR DESCRIPTION
**Pull request by jedwins1998**
Change `checkJobSetActive` function to check for JobSet conditions `JobSetFailed`, `JobSetComplete`, and `JobSetSuspended` as indicators that the JobSet is not active.

This fixes #530

### CODE REVIEW

**Comment by jedwins1998** [comment]
cc @danielvegamyhre

**Comment by jedwins1998** [comment]
Update as I realized the code I pushed did not match the definition I left on [this comment](https://github.com/kubernetes-sigs/jobset/issues/530#issuecomment-2065388816) in issue #531.

File: test/util/util.go [hunk 1]
```diff
@@ -119,7 +120,27 @@ func JobSetStartupPolicyNotFinished(ctx context.Context, k8sClient client.Client
...
+	var fetchedJS jobset.JobSet
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: js.Namespace, Name: js.Name}, &fetchedJS); err != nil {
+		return false, err
+	}
+
+	forbiddenTypes := []string{string(jobset.JobSetFailed), string(jobset.JobSetCompleted), string(jobset.JobSetSuspended)}
+	forbiddenStatuses := []metav1.ConditionStatus{metav1.ConditionTrue}
+
+	for _, c := range fetchedJS.Status.Conditions {
+		if slices.Contains(forbiddenTypes, c.Type) && slices.Contains(forbiddenStatuses, c.Status) {
```
Comment (danielvegamyhre) [inline test/util/util.go:139]: All of these conditions must have a status of true in order to be failed/completed/suspended, so let's just directly check `slices.Contains(forbiddenTypes, c.Type) && c.Status == metav1.ConditionTrue`

Comment (jedwins1998) [reply test/util/util.go]: Updated the code.

**Comment by danielvegamyhre** [comment]
/approve

Thanks!

**Pull request merged**